central/image/service/service_impl.go

```diff
@@ -296,6 +297,38 @@ func (s *serviceImpl) ScanImageInternal(ctx context.Context, request *v1.ScanIma
 	return internalScanRespFromImage(img), nil
 }
 
+// updateImageFromRequest will update the name of existing image with the one from the request
+// if the names differ and the metadata for the existing image was unable to be pulled previously.
+func updateImageFromRequest(existingImg *storage.Image, reqImgName *storage.ImageName) {
+	if !features.UnqualifiedSearchRegistries.Enabled() || reqImgName == nil {
+		// The need for this behavior is associated with the use of unqualified search
+		// registries or short name aliases (currently), if the feature is disabled
+		// do not modify the name.
+		return
+	}
+
+	if existingImg.GetMetadata() != nil {
+		// If metadata exists, then the existing image name is likely valid, no update needed.
+		return
+	}
+
+	baseImgName := existingImg.GetName()
```

Comment:
nit: I got confused regarding `base vs existing`; maybe best to keep things similar and rename this to `existingImgName`.